# The Basics of moving and controlling. 

In [1]:
import sys
#the uArm application is inside of the directory 
sys.path.append('..')
from uarm.wrapper import SwiftAPI


swift = SwiftAPI(port="COM4", callback_thread_pool_size=1)

## Set the arm back to the center. 

In [2]:
swift.reset()

## First, where is the arm right now?


In [3]:
swift.get_position()

[200.0, 0.0, 150.0]

The angle here is in `[x,y,z]`

If you were the arm, with the cables connected to your back and looking forward.
* `x` = left and right 
* `y` = the arm stretched outward or contracted inward
* `z` = up and down. 

In [4]:
swift.set_position(x=100, y=10, z=200)

In [5]:
swift.get_position()

[200.0, 0.0, 150.0]

In [6]:
swift.set_position(x=100, y=100, z=100)

In [7]:
swift.get_position()

'TIMEOUT'

# Moving individual components

## by ANGLE

servo_id：
* 0: BOTTOM - "Rotates the arm on the base"
* 1: LEFT - "Stretches the arm outward"
* 2: RIGHT - 
* 3: HAND - "Gripper wrist"

Note that 3(wrist) does not print out in the `get_servo_angle` 

when setting any angle, remember the following: 
* 0 is the far left. 
* 90 in center
* 180 is far right

However you have to think about the position of the servo, not the position of the arm. For example on the bottom (base) servo, going from 90 degrees to zero moves servo counter clockwise. However you are facing the other way so the servo turns clockwise and the arm moves to the right. 

In [8]:
swift.set_servo_angle(0,100)

In [9]:
swift.set_servo_angle(1,0)

In [10]:
swift.set_servo_angle(2,180)

In [11]:
swift.set_servo_angle(3,0)

In [12]:
swift.get_servo_angle()

[135.0, 123.75, 18.22]

In [13]:
swift.get_servo_angle(servo_id=1)

123.75

## by POSITION

Parameter：
* x：X axis value, default value: None, use the X value of last movement or initial value 150.
* y：Y axis value, default value: None, use the Y value of last movement or initial value 0.
* z：Z axis value, default value: None, use the Z value of last movement or initial value 150.
* speed：speed value, default value: None, use the speed value of last movement for default speed 1000.
* relative：is/isn’t relative movement, default value: False
* wait：do/don’t wait for return, default value: False


In [14]:
swift.set_position()

### 4th motor (the wrist arm of the grabber)

* **Angle** A value between 0 and 180. 0 = far left, 180 = far right, 90 = center

In [15]:
swift.set_wrist(10)

### the gripper (if attached)

`True` = closes the gripper
`False` = opens the gripper

In [16]:
swift.set_gripper(True)

'TIMEOUT'

##  QA


In [17]:
swift.get_temperature()

{'current_temperature': 0.0, 'target_temperature': 0.0}

In [18]:
swift.get_power_status()

True

In [19]:
swift.get_device_info()

{'device_type': 'SwiftPro',
 'hardware_version': '3.3.1',
 'firmware_version': '3.2.0',
 'api_version': '3.2.0',
 'device_unique': '78DB2FB28F96'}

In [20]:
swift.get_mode()

0

In [21]:
swift.get_servo_attach()

True

## Cleanup: disconnect

In [22]:
swift.disconnect()